app/controllers/api/v1/customers_controller.rb

```diff
@@ -10,15 +10,29 @@ class CustomersController < Api::V1::BaseController
       skip_authorization_check only: :index
 
       before_action :authorize_action, only: [:show, :update, :destroy]
+      # Query parameters
+      before_action :set_display_customer_balance, only: [:index] # Always show balance for show
 
       def index
         @pagy, customers = pagy(search_customers, pagy_options)
-
-        render json: Api::V1::CustomerSerializer.new(customers, pagination_options)
+        render json: Api::V1::CustomerSerializer.new(
+          customers,
+          pagination_options.merge({
+                                     params: {
+                                       display_customer_balance: @display_customer_balance
+                                     }
+                                   })
+        )
```

Comment:
To answer your style question. I would format it like this:
```suggestion
        render json: Api::V1::CustomerSerializer.new(
          customers,
          pagination_options.merge(
            params: { display_customer_balance: @display_customer_balance }
          )
        )
```

Well, actually, in this case I would simplify it further:


```suggestion
        options = pagination_options.merge(
          params: { display_customer_balance: @display_customer_balance }
        )
        render json: Api::V1::CustomerSerializer.new(customers, options)
```

But, as said above, I would hope not to need this option at all.